### PR TITLE
[node-core-library] Improve performance of Import.resovlePackage.

### PR DESCRIPTION
**Pull request by iclanton**
Addresses the feedback in this comment: https://github.com/microsoft/rushstack/pull/2095/files#r478033549

### CODE REVIEW

File: libraries/node-core-library/src/Import.ts [hunk 1]
```diff
@@ -264,21 +263,26 @@ export class Import {
...
+    if (allowSelfReference) {
+      const ownPackage: IPackageDescriptor | undefined = Import._getPackageName(baseFolderPath);
       if (ownPackage && ownPackage.packageName === packageName) {
         return ownPackage.packageRootPath;
       }
     }
 
+    const { error } = PackageName.tryParse(packageName);
+    if (error) {
+      throw new Error(error);
```
Comment (D4N14L) [inline libraries/node-core-library/src/Import.ts:283]: I'm surprised there isn't a `PackageName.parse` method which would just do this.

File: common/changes/@rushstack/node-core-library/import-improvements_2022-12-09-01-39.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@rushstack/node-core-library",
+      "comment": "Improve the error message emitted when a path inside a package is passed to `Import.ResolvePackage`.",
```
Comment (octogonz) [inline common/changes/@rushstack/node-core-library/import-improvements_2022-12-09-01-39.json:5]: Capitalization should be `resolvePackage`

Comment (iclanton) [reply common/changes/@rushstack/node-core-library/import-improvements_2022-12-09-01-39.json:5]:
```suggestion
      "comment": "Improve the error message emitted when a path inside a package is passed to `Import.resolvePackage`.",
```

**Review by D4N14L — APPROVED** [review]

File: libraries/node-core-library/src/Import.ts (hunk 1 above)
Comment (iclanton) [reply libraries/node-core-library/src/Import.ts:283]: Lol actually there is. Switched to that.

**Pull request merged**